src/main/java/org/jenkinsci/plugins/GitLabOAuthUserDetails.java

```diff
@@ -6,21 +6,18 @@
 
 import org.acegisecurity.GrantedAuthority;
 import org.acegisecurity.userdetails.User;
-import org.acegisecurity.userdetails.UserDetails;
+
 
 /**
  * @author Mike
  *
  */
-public class GitLabOAuthUserDetails extends User implements UserDetails {
+public class GitLabOAuthUserDetails extends User {
 
-    /**
-     *
-     */
     private static final long serialVersionUID = 1709511212188366292L;
 
     public GitLabOAuthUserDetails(org.gitlab4j.api.models.User user, GrantedAuthority[] authorities) {
-        super(user.getUsername(), "", true, true, true, true, authorities);
+        super(user.getUsername(), user.getEmail(), true, true, true, true, authorities);
```

Comment:
Why was this changed?  I don't see any indication in the spotbugs output that it needs to be changed and I see no indication in any of the commit messages why it needs to be changed.  What did I miss?
```suggestion
        super(user.getUsername(), "", true, true, true, true, authorities);
```

Reply:
I reverted that change in b8e5404eb11b22fbe8c00647038f3ed89f849c20 .  If there is a reason to restore that change, please do so with an additional commit that includes a message describing the reason for the change.

Reply:
hard-coded values for the User constructor parameters might not be appropriate for all cases and may cause issues in the application. 
So I replaced them with the username and email properties from the org.gitlab4j.api.models.User object to initialize the User superclass. 
This ensures that the GitLabOAuthUserDetails class is properly initialized with the correct values from the org.gitlab4j.api.models.User object.
correct me if I am wrong here.

Reply:
Changing the behavior of the system as part of a spotbugs warnings improvement is a dangerous choice.  I can't accept it.  I've harmed users more than once when I thought that a "cleanup" step was safe, but later proved that users relied on the previous behavior.

You're assuming that the empty string argument is worse than replacing it with another value, yet there is no evidence in any issue report or in any other location that the users want that change or that users will be unharmed by that change.  Authentication and authorization are critical parts of the system.  If this change inadvertently breaks a user, they could be locked out of their system.

If the change is needed, then it needs to happen in another pull request so that there will be a clearly phrased changelog entry that describes the behavior change to users.  The impact of the change needs to be assessed by reviewing how the caller uses the second argument and what impact the change from empty string to email address will have.

Reply:
ok thanks for the explanation